### PR TITLE
fix(ingestion/glue): Add support for missing config options for profiling in Glue

### PR DESCRIPTION
**Pull request by sagar-salvi-apptware**
**Summary:**

1. **Glue Source Profiling Configuration:**
   - Configuration parameters include:
     - `enabled`: Flag to enable or disable profiling (default: `false`).
     - `profile_table_level_only`: Flag to enable profiling at the table level only, excluding column-level profiling (default: `false`).
     - `max_workers`: Number of worker threads to use for profiling, defaulting to 5 times the CPU count.

2. **Test Cases:**
   - Added new test cases for Glue source to verify the new profiling configuration.
   - Ensured that existing Glue source test cases are compatible with the new configuration.

3. **Documentation Update:**
   - Updated the `updating-datahub.md` file to document the breaking change related to the profiling configuration for Glue source under the "Breaking Changes" section.

**QA:**
- Validated the change locally to ensure its working as expected

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **New Features**
  - Introduced profiling support for AWS Glue data ingestion, including options to enable profiling and restrict it to table-level only.

- **Improvements**
  - Simplified profiling logic for enhanced readability and efficiency.

- **Tests**
  - Added comprehensive test cases to validate profiling functionalities within the AWS Glue data ingestion module.

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/aws/glue.py [hunk 1]
```diff
@@ -944,21 +948,22 @@ def get_profile_if_enabled(
...
-
-                    if self.source_config.profiling.partition_patterns.allowed(
-                        partition_spec
-                    ):
-                        yield self._create_profile_mcp(
-                            mce, table_stats, column_stats, partition_spec
-                        ).as_workunit()
-                    else:
-                        continue
+                with ThreadPoolExecutor(
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/aws/glue.py:951]: Do we make any API call or query as part of `_create_partition_profile_mcp` ? I didn't notice any. If we are not making any network calls, then `max_workers` config does not make sense and we should remove it from here.

Comment (sagar-salvi-apptware) [reply metadata-ingestion/src/datahub/ingestion/source/aws/glue.py:951]: @mayurinehate yes, we are not making any network calls i ll remove it.

File: metadata-ingestion/src/datahub/ingestion/source/aws/glue.py [hunk 2]
```diff
@@ -867,34 +868,39 @@ def _create_profile_mcp(
...
+                        self.source_config.profiling.mean
+                    ]
+                if self.source_config.profiling.median in column_params:
+                    column_profile.median = column_params[
+                        self.source_config.profiling.median
+                    ]
+                if self.source_config.profiling.stdev in column_params:
+                    column_profile.stdev = column_params[
+                        self.source_config.profiling.stdev
+                    ]
```
[the remark this answers is not in the document]
Comment (sagar-salvi-apptware) [reply metadata-ingestion/src/datahub/ingestion/source/aws/glue.py:902]: Not needed as we removed this _create_partition_profile_mcp fun

File: metadata-ingestion/tests/unit/test_glue_source.py [hunk 3]
```diff
@@ -641,3 +684,51 @@ def fake_schema_metadata(entity_urn: str) -> models.SchemaMetadataClass:
...
     )
+
+
+@pytest.mark.parametrize(
+    "platform_instance, mce_file, mce_golden_file",
+    [
+        (None, "glue_mces.json", "glue_mces_golden_profiling.json"),
+    ],
+)
+@freeze_time(FROZEN_TIME)
```
Comment (mayurinehate) [inline metadata-ingestion/tests/unit/test_glue_source.py:695]: Not clear to me why this function needs to be parameterised if there is only one combination input.

**Review by mayurinehate — APPROVED** [review]
Minor comment on tests. Otherwise LGTM

**Review by anshbansal — APPROVED** [review]

**Comment by anshbansal** [comment]
Failure in airflow unrelated

**Pull request merged**